test/cache_test.lua

```diff
@@ -0,0 +1,245 @@
+    t.assert_equals(res.raw_tuple[SHARD_FUNC_BODY_IDX], func_body_first)
+    local id, err = ddl.bucket_id('space', 42)
+    t.assert_equals(err, nil)
+    t.assert_type(id, 'number')
+    t.assert_equals(id, 43)
+
+    box.space._ddl_sharding_func
+       :update({'space'}, {{'=', SHARD_FUNC_BODY_IDX, func_body_second}})
+
+    local res = cache.internal.get('space')
+    t.assert(res)
+    t.assert(res.callable)
+    t.assert_type(res.callable, 'function')
+    t.assert_equals(res.callable(42), 44)
+    t.assert(res.raw_tuple)
+    t.assert_equals(res.error, nil)
+    t.assert_equals(res.raw_tuple[SPACE_NAME_IDX], 'space')
+    t.assert_equals(res.raw_tuple[SHARD_FUNC_NAME_IDX], nil)
+    t.assert_equals(res.raw_tuple[SHARD_FUNC_BODY_IDX], func_body_second)
+    local id, err = ddl.bucket_id('space', 42)
+    t.assert_equals(err, nil)
+    t.assert_type(id, 'number')
+    t.assert_equals(id, 44)
+end
+
+function g.test_cache_space_updated_name()
+    local sharding_func_first = 'sharding_func_first'
+    rawset(_G, sharding_func_first, function(key) return key + 1 end)
+    rawset(_G, 'sharding_func_second', function(key) return key + 2 end)
+    g.schema.spaces.space.sharding_func = sharding_func_first
+    local ok, err = ddl.set_schema(g.schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(ok, true)
+
+    local res = cache.internal.get('space')
+    t.assert(res)
+    t.assert(res.parsed_func_name)
+    t.assert_type(res.parsed_func_name, 'table')
+    t.assert_equals(res.parsed_func_name[1], 'sharding_func_first')
+    t.assert_equals(res.raw_tuple[SPACE_NAME_IDX], 'space')
+    t.assert_equals(res.raw_tuple[SHARD_FUNC_NAME_IDX], 'sharding_func_first')
+    local id, err = ddl.bucket_id('space', 42)
+    t.assert_equals(err, nil)
+    t.assert_type(id, 'number')
+    t.assert_equals(id, 43)
+
+    box.space._ddl_sharding_func
+       :update({'space'}, {{'=', SHARD_FUNC_NAME_IDX, 'sharding_func_second'}})
+
+    local res = cache.internal.get('space')
+    t.assert(res)
+    t.assert(res.parsed_func_name)
+    t.assert_type(res.parsed_func_name, 'table')
+    t.assert_equals(res.parsed_func_name[1], 'sharding_func_second')
+    t.assert_equals(res.raw_tuple[SPACE_NAME_IDX], 'space')
+    t.assert_equals(res.raw_tuple[SHARD_FUNC_NAME_IDX], 'sharding_func_second')
+    local id, err = ddl.bucket_id('space', 42)
+    t.assert_equals(err, nil)
+    t.assert_type(id, 'number')
+    t.assert_equals(id, 44)
+end
+
+function g.test_cache_nil_space()
+    local sharding_func_first = 'sharding_func_first'
+    rawset(_G, sharding_func_first, function(key) return key + 1 end)
+    rawset(_G, 'sharding_func_second', function(key) return key + 2 end)
+    g.schema.spaces.space.sharding_func = sharding_func_first
+    local ok, err = ddl.set_schema(g.schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(ok, true)
+
+    local res = cache.internal.get()
+    t.assert_equals(res, nil)
+end
```

Comment:
Added a small testcase to cover a case when space name is `nil`. Test makes a 100% coverage for module `ddl/cache.lua`. 

![Screenshot from 2022-01-25 12-53-01](https://user-images.githubusercontent.com/1151557/150954368-cfc4f79f-08cc-4009-b890-914d9f75adbf.png)


```lua
--- a/test/cache_test.lua
+++ b/test/cache_test.lua
@@ -230,3 +230,16 @@ function g.test_cache_space_updated_name()
     t.assert_type(id, 'number')
     t.assert_equals(id, 44)
 end
+
+function g.test_cache_nil_space()
+    local sharding_func_first = 'sharding_func_first'
+    rawset(_G, sharding_func_first, function(key) return key + 1 end)
+    rawset(_G, 'sharding_func_second', function(key) return key + 2 end)
+    g.schema.spaces.space.sharding_func = sharding_func_first
+    local ok, err = ddl.set_schema(g.schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(ok, true)
+
+    local res = cache.internal.get()
+    t.assert_equals(res, nil)
+end
```

Reply:
(Or we can remove the check — it is anyway not a valid input and I hope that API functions doing input validation good enough.)

Reply:
I prefer insurance, test added

Reply:
(Looks resolved, so I'm going to resolve this discussion.)